frontend/controller/actions/utils.js

```diff
@@ -310,7 +310,8 @@ export function groupContractsByType (contracts: Object): Object {
   const contractIDs = Object.create(null)
   if (contracts) {
     // $FlowFixMe[incompatible-use]
-    Object.entries(contracts).forEach(([id, { type }]) => {
+    Object.entries(contracts).forEach(([id, { references, type }]) => {
+      if (!references) return
```

Comment:
Could you add a comment here explaining under what circumstances `references` might not exist?

Reply:
Sure. Short story: for ephemeral references and 'internal' references (i.e., foreign keys)